csrc/scheduler/transpose.cpp

```diff
@@ -714,29 +718,124 @@ std::shared_ptr<TransposeParams> getTransposeHeuristics(
       ceilDiv((int64_t)params->tile_size1 * (int64_t)params->tile_size2, 32l);
   max_unroll_factor = std::min(max_unroll_factor, max_unroll_factor_block);
 
-  // Compute maximum vectorize factor that can be used
-  size_t vectorize_factor1 = max_unroll_factor;
-  size_t vectorize_factor2 = max_unroll_factor;
+  // Note: [Computing Vectorization Width for Transpose]
+  //
+  // With support of small transpose dimension (see Note [Supporting small
+  // transpose dimensions]), we need to consider the transformation applied on
+  // our tile to compute the safe vectorization width. e.g. For a simple
+  // transpose:
+  //    (i0, i1, i2, i3, i4) size (2, 3, 65536, 4, 7)
+  // -> (i3, i4, i2, i0, i1)
+  //
+  // transpose scheduler would apply transformation and tile on virtual
+  // innermost dimensions. ( (i0*i1*i2/2), (2*i3*i4)/32) So we need to use the
+  // size of the virtual innermost dimensions to compute our vectorization
+  // width. In the example above, we are looking at 2*3*65536/2, 2*4*7.
+  //
+  // Currently there's limitation on our iter domain mapping. Since we can only
+  // do it on rfactor/root domain, we cannot map across `split` domains. So the
+  // example above will only have vectorization size of 2 and 4 repsectively for
+  // the merge virtual innermost dimensions, rather than considering the split
+  // and merged i2/2 & 2.
```

Comment:
Is this still the case even with https://github.com/NVIDIA/Fuser/pull/753?

Reply:
Yes that's still the case.

We need to further refactor the mapping thing in order to support that.

Reply:
Isn't this automatically handled in `projectId`?

Reply:
Hmmm. I wasn't totally sure about this one. I was still trying to have projectId to map ids to `rfacotr_domain` of the reference. I think previously I was struggling with `getProjectedExtent` as well as propogation. But that might have been fixed with us changing addProjectedExtent to before the projectId call.

I think you are saying it's not necessary and propagation should still be able to handle that. Let me give this another shot to see if it would just work.

Reply:
I think this is still the case here.

Since we project leaf domain to rfactor_domain before propagating them in `ContiguousInnerDimensionsMapper`, so for split leaf domains, they would simply vanish before we get the chance to actually propagate them.

This is also verified in the test added in this PR:
`TransposeSplitAggregatedVectorizationWidth_CUDA`

Do still plan to support this, but it's going to be another refactor on the mapper.

Reply:
Since we have this `auto projected_rfactor = projectId(filtered_ids, rfactor_domain);` before propagation.
```
T1_g[ iS5{i3}, iS6{( ceilDiv(i2, 5) )}, iS7{5}, iS3{i0} ]
 root domain : (iS3{i0}, iS4{i2}, iS5{i3})
 rfactor domain : (iS5{i3}, iS4{i2}, iS3{i0})
 contiguity: t t t
  Split: iS4{i2} by factor 5 -> iS6{( ceilDiv(i2, 5) )}, iS7{5}, start offset: 0, stop offset: 0
 leaf domain : (iS5{i3}, iS6{( ceilDiv(i2, 5) )}, iS7{5}, iS3{i0})
```

```
(gdb) print filtered_ids
$9 = std::vector of length 2, capacity 2 = {0x55555acf9050, 0x555559df9970}
(gdb) print projected_rfactor
$10 = std::vector of length 1, capacity 2 = {0x555559df9970}
(gdb) print filtered_ids[0]->toString(0)
$5 = "iS7{5}"
(gdb) print filtered_ids[1]->toString(0)
$6 = "iS3{i0}"
(gdb) print projected_rfactor[0]->toString(0)
$4 = "iS3{i0}"
```
So the split dimension `iS7{5}` is lost when we project it back.

Reply:
Ah, OK, so, that means we can't figure out the vectorization factor for tensors that already have non-rfactor leaf domains, right? 

Can this be solved by projecting back from the rfactor to leaf domains?

Reply:
I think so. It's just a refactor to have the propagator to use leaf instead of rfactor. I just don't know enough about how easy the new propagation is going to be, but would love to do this in next round. :office_worker: